src/elements/OcFile.vue

```diff
@@ -26,49 +26,88 @@ export default {
       default: "folder",
     },
     /**
-     * A file object with necessary property `name` and an optional property `extension`.
+     * The name of a file.
      * Please note that the name shall not hold the extension
      */
-    file: {
-      type: Object,
+    name: {
+      type: String,
       required: true,
+      default: " ",
+    },
+    /**
+     * The extension of a file`.
+     */
+    extension: {
+      type: String,
+      required: false,
+      default: " ",
     },
   },
   methods: {
     onClick() {
       /**
        * This event is emitted when the user clicks a file
-       * TODO https://vuejs.org/v2/guide/components-custom-events.html#Binding-Native-Events-to-Components
        * @type {string}
        */
-      this.$emit("oc-file:click", this.file)
+      this.$emit("click", this.file)
```

Comment:
ok, so @felixheidecke already mentioned that we should come up with an event naming scheme.
1. The official docs recommend to **always use kebab-case for event names**. See https://stackoverflow.com/questions/42441952/vue-js-custom-event-naming
2. I don't know if we need to namespace *our* events or if there is a built in mechanism to add only to click events on an oc-icon but not on oc-file clicks ... after reading https://stackoverflow.com/a/53107274 I think we may not need namespaces if the component throws an event ... hm no ... but we do need global events eg for the file actions .... hm no that is handled by the outer vue component ... whatever ... someone with more in depth know how should share his insight.

Reply:
1. use existing event names where it is reasonable
2. yes - no namespace needed

Reply:
maybe we do ... "oc-file:click" is picked up in https://github.com/owncloud/owncloud-design-system/blob/master/src/templates/Files.vue#L160

You would have to change that as well, but now it is 'just' `click`... but anything might fire a click ...

Reply:
@DeepDiver1975 https://github.com/owncloud/owncloud-design-system/blob/master/src/templates/Files.vue#L160 needs to use `click` instead of `oc-file:click` as well.